src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java

```diff
@@ -266,4 +270,114 @@ public StringValueResponse terminateEvent(@PathVariable("procID") final Integer
     procurementEventService.terminateEvent(procId, eventId, type.getTerminationType(), principal);
     return new StringValueResponse("OK");
   }
+
+  @GetMapping("/{eventID}/responses/export")
+  public  ResponseEntity<StreamingResponseBody> exportAllResponses(
+          @PathVariable("procID") final Integer procId, @PathVariable("eventID") final String eventId,
+          HttpServletResponse response, final JwtAuthenticationToken authentication){
+
+    var principal = getPrincipalFromJwt(authentication);
+    log.info("getDocumentSummaries invoked on behalf of principal: {}", principal);
+
+    List<SupplierAttachmentResponse> supplierAttachmentResponseList=procurementEventService.getSupplierAttachmentResponses(procId, eventId);
+
+    StreamingResponseBody streamResponseBody =
```

Comment:
This block producing the streaming response body is duplicated across the two new endpoint methods - can some or all of it be moved into a separate method and shared?

Reply:
done